moto/ec2/models.py

```diff
@@ -311,26 +319,65 @@ def __init__(
         #   returns groups for both self and the attached instance.
         self._group_set = []
 
+        if self.subnet.ipv6_cidr_block_associations:
+            association = list(self.subnet.ipv6_cidr_block_associations.values())[0]
+            subnet_ipv6_cidr_block = association.get("ipv6CidrBlock")
+            if kwargs.get("ipv6_address_count"):
+                while True:
+                    if len(self.ipv6_addresses) == kwargs.get("ipv6_address_count"):
```

Comment:
I'm not an expert in EC2 at all, but could this ever be true?
`len(self.ipv6_addresses) > ipv6_address_count`

If that condition is met, we'd have an infinite loop on our hands.
This might be a better way:
```
while len(self.ipv6_addresses) < kwargs.get("ipv6_address_count"):
```